tests/lib/rules/jsx-curly-spacing.js

```diff
@@ -83,6 +83,88 @@ ruleTester.run('jsx-curly-spacing', rule, {
     options: ['never'],
     parserOptions: parserOptions,
     parser: 'babel-eslint'
+  }, {
+    code: '<App {...bar} />;',
+    parserOptions: parserOptions
+  }, {
+    code: '<App {...bar} />;',
+    options: ['never'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App { ...bar } />;',
+    options: ['always'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App { ...bar } />;',
+    options: ['always', {allowMultiline: false}],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<App {',
+      '...bar',
+      '} />;'
+    ].join('\n'),
+    options: ['always'],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<App {',
+      '...bar',
+      '} />;'
+    ].join('\n'),
+    options: ['always'],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<App {',
+      '...bar',
+      '} />;'
+    ].join('\n'),
+    options: ['never'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App {...bar/* comment */} />;',
+    options: ['never'],
+    parserOptions: parserOptions,
+    parser: 'babel-eslint'
+  }, {
+    code: '<App foo={bar} {...baz} />;',
+    parserOptions: parserOptions
+  }, {
+    code: '<App foo={bar} {...baz} />;',
+    options: ['never'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App foo={ bar } { ...baz } />;',
+    options: ['always'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App foo={ bar } { ...baz } />;',
+    options: ['always', {allowMultiline: false}],
+    parserOptions: parserOptions
+  }, {
+    code: '<App foo={{ bar:baz }} {...baz} />;',
+    options: ['never'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App foo={ {bar:baz} } { ...baz } />;',
+    options: ['always'],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<App foo={',
+      'bar',
+      '} {',
+      '...bar',
+      '}/>;'
+    ].join('\n'),
+    options: ['always'],
+    parserOptions: parserOptions
+  }, {
+    code: '<App foo={bar/* comment */} {...baz/* comment */} />;',
+    options: ['never'],
+    parserOptions: parserOptions,
+    parser: 'babel-eslint'
```

Comment:
this test case was added in https://github.com/yannickcr/eslint-plugin-react/pull/584, and it seems that the issue wasn't related with babel-eslint parse, so I think it is safe to remove them both